cluster-autoscaler/processors/nodegroupset/aws_nodegroups.go

```diff
@@ -21,16 +21,16 @@ import (
 )
 
 // IsAwsNodeInfoSimilar adds AWS specific node labels to the list of ignored labels.
-func IsAwsNodeInfoSimilar(n1, n2 *schedulernodeinfo.NodeInfo) bool {
+func IsAwsNodeInfoSimilar(n1, n2 *schedulernodeinfo.NodeInfo, ignoredLabels map[string]bool) bool {
```

Comment:
please do not modify input parameter.
Copy the map and extend with standard awsIngornedLabels before passing to `IsNodeInfoSimilar`

Reply:
Actually I have another idea. It does not seem best idea to pass `ignoredLabels` to each call of comparator.
The passed labels will be the same ever time. And each call we do the same merging of maps

I suggest following apporach:
 * replace static Is*NodeInfoSimilar functions with builders.
   * The builder would be parametrized with `extraIgnoredLabels` and would merge the `Basic`, `extra`, and `cloud-provider specific` ignored labels. 
   * The generated comparator would just take two node `*NodeInfo` parameters (no ignoredLabels map)
   * E.g For AWS we create `CreateAwsNodeInfoSimilarFunction() func(n1, n2 *NodeInfo) bool`. Similarly for Azure. The returned function would call out to standard `IsNodeInfoSimilar(n1, n2 *NodeInfo, ignoredLabels map[string]bool ignoredLabels)`.
   * If neither AWS nor Azure is used, then comparator would be created with `CreateStandardNodeInfoSimilarFunction` which would just merge `Basic` and `extra` ignored node labels and also return function which calls out to `IsNodeInfoSimilar(n1, n2, ignoredLabels)`

I think that approach would be cleaner. WDYT?

Reply:
Yep. Currently rewriting this PR to use this approach.